### ISSUE TITLE
/gldevice:Vulkan ignored under linux.

### ISSUE THREAD

**Issue by tjhexf** [bluemagic123/tModLoader#3514]
### Version

Latest 1.4-stable

### OS

Linux

### Platform

Steam

### This bug affects

Gameplay as a Player

### Description

After appending /gldevice:Vulkan to the launch options, it still refuses to launch in Vulkan mode, instead defaulting to openGl, a behavior not observed under vanilla terraria, which correctly launches with Vulkan.

### Log File

[client.log](https://github.com/tModLoader/tModLoader/files/11644674/client.log)

### Steps to reproduce

Append /gldevice:Vulkan to launch options.

### Expected Behavior

Game launches using Vulkan renderer.

### Actual Behavior

Still uses Opengl.

### Reproduction frequency

Every time.

### Additional Information

Fully updated archlinux system as of current writing date.

**Comment by tjhexf** [comment]
Any attempts, be it launching from the command line or steam, still do not respect /gldevice:Vulkan.

**Comment by Solxanich** [comment]
Please confirm the following:
1) Does this behavior persist after removing `-mono ...` ? tModLoader does not use mono, and that seems to be affecting args adversely
2) Please post additional logs (native, client.log)  after removing the `-mono`

I'm surprised it isn't working as is on linux, we've tested previously fine, and not sure why vulkan would be overridden

**Comment by tjhexf** [comment]
> Please confirm the following:
> 
>     1. Does this behavior persist after removing `-mono ...` ? tModLoader does not use mono, and that seems to be affecting args adversely
> 
>     2. Please post additional logs (native, client.log)  after removing the `-mono`
> 
> 
> I'm surprised it isn't working as is on linux, we've tested previously fine, and not sure why vulkan would be overridden

Yes, it still does not work, even without the mono option. I will append the logs below:



[Natives.log](https://github.com/tModLoader/tModLoader/files/11653139/Natives.log)
[client.log](https://github.com/tModLoader/tModLoader/files/11653141/client.log)

**Comment by tjhexf** [comment]
Interestingly enough, none of the logs seem to imply any error or exception at any point. It seems to be quietly ignoring vulkan.

**Comment by tjhexf** [comment]
@Solxanich TmodLoader's source code seems to supply its own patches to fna3d, is it possible that could be affecting the vulkan renderer on linux?

**Comment by tjhexf** [comment]
Update:

Both the steam version and latest release from github do not respect Vulkan launch options.

**Comment by tjhexf** [comment]
Update, manually setting the environment value:
FNA3D_FORCE_DRIVER=Vulkan

makes fna3d respect Vulkan, in both steam and github releases.

Weirdly enough, it seems somewhere in the launch process /gldevice is being casually ignored, but still usable with environment values.

**Comment by Chicken-Bones** [comment]
Thanks for the info. I'm willing to fix this, but can you tell me, which piece of documentation told you that `/gldevice:Vulkan` should work?

**Comment by Solxanich** [comment]
Apologies, had some internet challenges last few days.
https://github.com/FNA-XNA/FNA/wiki/7:-FNA-Environment-Variables
`/gldevice:OpenGL` is a commonly supported argument on windows FNA. 
When found, FNA exports the variable mentioned (=OpenGL) for usage, prior to init. 

On Terraria, this works as the argument is preserved by -mono and processed 

On tModLoader, the /gl argument does not make it to the final cleaned args, and even if we add the -mono, it still not processed. 

I've been trying to find a chance to look in FNA code for that exact export, just haven't had a good connection to do so

**Comment by tjhexf** [comment]
> Thanks for the info. I'm willing to fix this, but can you tell me, which piece of documentation told you that `/gldevice:Vulkan` should work?

Oh, the FNA3D documentation. It's the user standard way of switching renderers. It works properly on terraria aswell as other fna3d games.

**Comment by Solxanich** [comment]
> Update, manually setting the environment value: FNA3D_FORCE_DRIVER=Vulkan
> 
> makes fna3d respect Vulkan, in both steam and github releases.
> 
> Weirdly enough, it seems somewhere in the launch process /gldevice is being casually ignored, but still usable with environment values.

The real confusing part is that the code path is identical on Windows, and it is working just fine there. 

This is the FNA code: 
If gldevice exists, use it to set FNA3d_FORCE_DRIVER
![image](https://github.com/tModLoader/tModLoader/assets/59670736/da3710b4-541f-44ba-aa7d-ca1fb39ed1ba)

And from your log, we know that `new LaunchParameters` should have `/gldevice:Vulkan` but it's not showing up.
![image](https://github.com/tModLoader/tModLoader/assets/59670736/f7b67d5f-2bc1-43f2-94eb-2c12172fbefc)

My working theory is some files are missing for Vulkan support for tModLoader... but all of those should be there

**Comment by tjhexf** [comment]
Updating about a year later. Issue does not seem present anymore, tmodloader preserves the Vulkan option
